src/BCEngineMCMC.cxx

```diff
@@ -1189,8 +1256,10 @@ bool BCEngineMCMC::LoadMCMC(TTree* mcmcTree, TTree* parTree, bool loadObservable
     fMCMCTreeLoaded = false;
     fMCMCTreeReuseObservables = loadObservables;
 
-    if (!mcmcTree || !parTree)
+    if (!mcmcTree || !parTree) {
+        BCLog::OutError("BCEngineMCMC::LoadMCMC : empty trees provided");
```

Comment:
I agree. But the function was created with the original `BAT`ish return value to show success, instead of the proper throwing that C++ should have. If I throw instead of returning `false`, then the return value becomes meaningless and should changed to `void`. If you agree we can change the return value, I'm happy to `throw`.

Reply:
Yes, please throw. Nobody seems to check error codes

Reply:
Oh, remember when they actually returned `int` instead of `bool`?

Reply:
> Oh, remember when they actually returned int instead of bool?

and `0` indicated a failure? puhh I don't want to remember